homework02/rudolf/string.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+#include <iostream>
+#include <memory>
+#include <algorithm>
+
+std::size_t strlen(const char* chars);
+
+class String
+{
+public:
+	String();
+
+	String(const char* chars,std::size_t capacity);
+
+	String(const char* chars):String(chars,strlen(chars)){};
```

Comment:
A few things on this line:
  * Nice that you used a delegating ctor
  * The ctor you are calling takes capacity and you are passing the size, which makes it off by one byte
  * The last `;` at the end of the line is not needed / can potentially cause a warning

Also, this style is preferred 

```
String(consrt char* str) : 
    String(str, std::strlen(str) + 1)
{}
```
It would be better to define it in the cc file though (then you dont need to include cstring here!)